mesh/ncmesh.cpp

```diff
@@ -1651,8 +1940,9 @@ void NCMesh::Refine(const Array<Refinement>& refinements)
    ref_stack.Reserve(refinements.Size());
    for (int i = refinements.Size()-1; i >= 0; i--)
    {
-      const Refinement& ref = refinements[i];
-      ref_stack.Append(Refinement(leaf_elements[ref.index], ref.ref_type));
+      Refinement ref = refinements[i];  // Copy
+      ref.index = leaf_elements[ref.index];
+      ref_stack.Append(ref);
```

Comment:
Why the change to an extra line? I think this also gives up the possibility of move semantics (though I doubt Array uses them).

Reply:
The purpose is to copy the 3 scaling factors in the array `s`. The constructor does not take this array. Originally, I wrote a constructor with optional arguments `sx, sy, sz`, which we decided was too awkward.